docs/editor/intellisense.md

```diff
@@ -0,0 +1,156 @@
+
+Below are the most popular language extensions in the [Marketplace](https://marketplace.visualstudio.com/vscode). Click on an extension tile below to read the description and reviews to decide which extension is best for you. 
+
+> **Tip:** For configuring and troubleshooting JavaScript IntelliSense see [here](/docs/languages/javascript#configuring-intellisense).
+
+<div class="marketplace-extensions-languages"></div>
+
+## IntelliSense Features
+
+If a language service knows possible completions, the IntelliSense suggestions will pop up as you type. If you continue typing characters, the list of members (variables, methods, etc.) is filtered to include only members containing your typed characters. Pressing `kbstyle(Tab)` or `kbstyle(Enter)` will insert the selected member. 
+
+You can trigger IntelliSense in any editor window by typing `kb(editor.action.triggerSuggest)` or by typing a trigger character (such as `.` in JavaScript). 
+
+![intellisense in package json](images/intellisense/intellisense_packagejson.gif)
+
+> **Tip:** The suggestions filtering supports CamelCase so you can type the upper case letters of a method name to limit the suggestions. For example, "cra" will quickly bring up "createApplication".
+
+> **Tip:** You can turn off IntelliSense while you type. See [Customize IntelliSense below](#_customize_intellisense).
+
+In addition, you can see **quick info** for each method (as provided by the language service). 
+
+![quick info](images/intellisense/quick_outline.png)
+
+After choosing a method you are provided with **parameter info**. 
+
+![paramater info](images/intellisense/paramater_info.png)
+
+
+## Suggestion Priority
+
+The order and the icon to the left of the word show the IntelliSense suggestion's priority.  
+
+> **Higher suggestion priority = higher order and a more detailed icon**
+
+This image shows the result of triggering IntelliSense on `express` a popular [Node.js module](https://expressjs.com/). 
+
+![image showing intellisense icons](images/intellisense/intellisense_icons.png)
+
+Methods, variables, and objects tend to be high priority. Simple word completion, tend to be lower priority. When the priority level is equivalent, ordering is done alphabetically. 
+
+|       |         |
+| ----- | ------- |
+| ![purple cube for a method](images/intellisense/method_icon.png) | Method or Function |
+| ![blue cuboid for a variable](images/intellisense/variable_icon.png) | Variable | 
+| ![blue circles connected by a blue line for an object](images/intellisense/object_icon.png) | Object |
+| ![a square with a small fold in the top left corner indicates word completion](images/intellisense/word_completion_icon.png) | Word Completion |
+| ![a square with a small fold in the top left corner indicates string completion](images/intellisense/string_completion_icon.png) | String completion |
+
```

Comment:
Should string completion be snippet completion. Snippets are a lot more than simple strings.